src/main/java/com/google/api/codegen/config/LongRunningConfig.java

```diff
@@ -152,18 +144,71 @@ private static LongRunningConfig createLongRunningConfigFromProtoFile(
       return null;
     }
 
-    // Use default retry settings.
-    Duration initialPollDelay = Duration.ofMillis(LRO_INITIAL_POLL_DELAY_MILLIS);
-    Duration maxPollDelay = Duration.ofMillis(LRO_MAX_POLL_DELAY_MILLIS);
-    Duration totalPollTimeout = Duration.ofMillis(LRO_TOTAL_POLL_TIMEOUT_MILLS);
-
-    return new AutoValue_LongRunningConfig(
-        ProtoTypeRef.create(returnType),
-        ProtoTypeRef.create(metadataType),
-        initialPollDelay,
-        LRO_POLL_DELAY_MULTIPLIER,
-        maxPollDelay,
-        totalPollTimeout);
+    LongRunningConfig.Builder builder =
+        getGapicConfigLroRetrySettingsOrDefault(diagCollector, longRunningConfigProto);
+
+    return builder
+        .setReturnType(ProtoTypeRef.create(returnType))
+        .setMetadataType(ProtoTypeRef.create(metadataType))
+        .build();
+  }
+
+  /**
+   * Creates an instance of LongRunningConfig.Builder based on a method's GAPIC config's LRO polling
+   * settings.
+   */
+  private static LongRunningConfig.Builder getGapicConfigLroRetrySettingsOrDefault(
+      DiagCollector diagCollector, LongRunningConfigProto longRunningConfigProto) {
+    if (longRunningConfigProto.equals(LongRunningConfigProto.getDefaultInstance())) {
+      return newBuilder()
+          .setInitialPollDelay(LRO_INITIAL_POLL_DELAY_MILLIS)
+          .setPollDelayMultiplier(LRO_POLL_DELAY_MULTIPLIER)
+          .setMaxPollDelay(LRO_MAX_POLL_DELAY_MILLIS)
+          .setTotalPollTimeout(LRO_TOTAL_POLL_TIMEOUT_MILLS);
+    }
+
+    Duration initialPollDelay =
+        Duration.ofMillis(longRunningConfigProto.getInitialPollDelayMillis());
+    if (initialPollDelay.compareTo(Duration.ZERO) < 0) {
+      diagCollector.addDiag(
+          Diag.error(
+              SimpleLocation.TOPLEVEL,
+              "Initial poll delay must be provided and set to a positive number: '%s'",
+              longRunningConfigProto.getInitialPollDelayMillis()));
+    }
+
+    double pollDelayMultiplier = longRunningConfigProto.getPollDelayMultiplier();
+    if (pollDelayMultiplier <= 1.0) {
```

Comment:
This causes a failure if pollDelayMultiplier == 1.0, but the comment in the error says that 1.0 is valid - they should be consistent. (I realize this was just taken from below, but let's fix it!)